conceptual/Npgsql/release-notes/8.0.md

```diff
@@ -1,29 +1,33 @@
+
+## Breaking changes
 
-```c#
-var builder = new NpgsqlDataSourceBuilder("<connection string>");
-builder.UseSystemTextJson();
-await using var dataSource = builder.Build();
-```
+### <a name="dynamic-optin">JSON POCO and other dynamic features now require an explicit opt-in
 
-Or, if you're using the legacy global type mapper, add the following at the start of your application, before any Npgsql operations are performed:
+Npgsql 8.0 is fully compatible with NativeAOT and trimming (see above). While most driver capabilities have been made to work in those profiles, certain features involve dynamic coding practices and are incompatible with NativeAOT and/or trimming - at least for now. As a result, these features now require explicit opt-ins (annotated to be incompatible with NativeAOT/trimming), which you must add either on your <xref:Npgsql.NpgsqlDataSourceBuilder> or on <xref:Npgsql.NpgsqlConnection.GlobalTypeMapper?displayProperty=nameWithType>:
 
-```c#
-NpgsqlConnection.GlobalTypeMapper.UseSystemTextJson();
-```
+PostgreSQL type                          | Default .NET type
+---------------------------------------- | --------------------------
+JSON POCO mapping, JsonNode and subtypes | <xref:Npgsql.INpgsqlTypeMapperExtensions.EnableDynamicJsonMappings>
+Unmapped enums, ranges, multiranges      | <xref:Npgsql.INpgsqlTypeMapperExtensions.EnableUnmappedTypes>
+Read PostgreSQL records as .NET tuples   | <xref:Npgsql.INpgsqlTypeMapperExtensions.EnableRecordsAsTuples>
 
-This change was done for several reasons. For one, as part of a push to improve the trimmability of Npgsql (for smaller sizes), the opt in ensures that System.Text.Json is only brought in for applications which use it, and gets trimmed out otherwise. In addition, the new opt-in API allows specifying configuration options, such as [JsonSerializerOptions](https://learn.microsoft.com/dotnet/api/system.text.json.jsonserializeroptions); this wasn't previously possible.
+Existing code using the above features will start throwing exceptions after upgrading to Npgsql 8.0; the exceptions provide explicit guidance on how to add the opt-ins.
 
-## Contributors
+### Plugin APIs have been changed for NativeAOT/trimming support
 
-Thank you very much to the following people who have contributed to the individual 8.0.x. releases.
+As part of the effort to make Npgsql compatible with NativeAOT and trimming, the plugin API was changed in fundamental, breaking ways. Although this API never had the stability guarantees of a true public API (it was and still is in an Internal namespace), external plugins were developed with it which will require adjustments.
```

Comment:
```suggestion
As part of the effort to make Npgsql compatible with NativeAOT and trimming, the plugin API was changed in fundamental, breaking ways. Although this API never had the stability guarantees of a true public API (it was and still is in an Internal namespace), external plugins which were developed with it will require adjustments.
```

Reply:
Maybe add a small note that we're still working towards more changes in this area and that it's advised to begin refactors after the 8.0 release.